static/app/components/events/interfaces/sourceMapsDebuggerModal.tsx

```diff
@@ -0,0 +1,1216 @@
+        >
+          {t('Was this helpful?')} <IconMegaphone size="xs" />
+        </Link>
+      </Footer>
+    </Fragment>
+  );
+}
+
+function getDebugIdProgress(sourceResolutionResults: SourceResolutionResults) {
+  let debugIdProgress = 0;
+  if (sourceResolutionResults.sdkDebugIdSupport === 'full') {
+    debugIdProgress++;
+  }
+  if (sourceResolutionResults.stackFrameDebugId !== null) {
+    debugIdProgress++;
+  }
+  if (sourceResolutionResults.uploadedSourceFileWithCorrectDebugId) {
+    debugIdProgress++;
+  }
+  if (sourceResolutionResults.uploadedSourceMapWithCorrectDebugId) {
+    debugIdProgress++;
+  }
+  return {debugIdProgress, debugIdProgressPercent: debugIdProgress / 4};
+}
+
+function getReleaseProgress(sourceResolutionResults: SourceResolutionResults) {
+  let releaseProgress = 0;
+  if (sourceResolutionResults.releaseName !== null) {
+    releaseProgress++;
+  }
+  if (sourceResolutionResults.uploadedSomeArtifactToRelease) {
+    releaseProgress++;
+  }
+  if (sourceResolutionResults.sourceFileReleaseNameFetchingResult === 'found') {
+    releaseProgress++;
+  }
+  if (sourceResolutionResults.sourceMapReleaseNameFetchingResult === 'found') {
+    releaseProgress++;
+  }
+  return {releaseProgress, releaseProgressPercent: releaseProgress / 4};
+}
+
+function getScrapingProgress(sourceResolutionResults: SourceResolutionResults) {
+  let scrapingProgress = 0;
+  if (sourceResolutionResults.sourceFileScrapingStatus.status === 'found') {
+    scrapingProgress++;
+  }
+  if (sourceResolutionResults.sourceMapScrapingStatus.status === 'found') {
+    // We give this step a relative weight of 4/5ths because this is actually way
+    // harder than step 1 and we want do deprioritize this tab over the others
+    // because the scraping process comes with a few downsides that aren't immediately
+    // obvious.
+    scrapingProgress += 4;
+  }
+  return {scrapingProgress, scrapingProgressPercent: scrapingProgress / 5};
+}
+
+function CheckListItem({children, title, status}: PropsWithChildren<CheckListItemProps>) {
+  return (
+    <ListItemContainer>
+      <CheckMarkContainer>
+        {
+          {
+            none: <IconCircle size="md" color="gray200" />,
+            checked: <IconCheckmark size="md" color="green300" isCircled />,
+            alert: <IconWarning size="md" color="yellow300" />,
+            question: <IconQuestion size="md" color="gray300" />,
+          }[status]
+        }
+        <Line className="source-map-debugger-modal-checklist-line" />
+      </CheckMarkContainer>
+      <ListItemContentContainer>
+        <ListItemTitleWrapper>
+          <ListItemTitle status={status}>{title}</ListItemTitle>
+        </ListItemTitleWrapper>
+        {children}
+      </ListItemContentContainer>
+    </ListItemContainer>
+  );
+}
+
+function InstalledSdkChecklistItem({
+  sourceResolutionResults,
+  setActiveTab,
+}: {
+  setActiveTab: React.Dispatch<
+    React.SetStateAction<'release' | 'debug-ids' | 'fetching'>
```

Comment:
l: Considering this is the third place we have this string union I'd rather just use a string enum and make the checks like `activeTab === DebuggerTab.DEBUG_IDS ? theme.purple300 : theme.gray300`

Reply:
I extracted the union into a type which pretty much does the same thing but is a bit cleaner to look at. (For some reason I don't like enums)